packages/tocco-ui/src/FormattedValue/typeFormatters/MultiSelectFormatter.js

```diff
@@ -6,29 +6,32 @@ import Typography from '../../Typography'
 export const MultiSeparator = () => ', '
 
 const MultiSelectFormatter = ({value, options, breakWords}) => {
-  return (
-    <Typography.Span breakWords={breakWords}>
-      {value && value.length > 0
-        ? value
-            .map(v =>
-              options && options.DetailLink ? (
-                <span onClick={e => e.stopPropagation()}>
-                  <options.DetailLink entityKey={v.key}>{v.display}</options.DetailLink>
-                </span>
-              ) : (
-                v.display
-              )
-            )
-            .reduce((prev, curr, idx) => [prev, <MultiSeparator key={'sep' + idx} />, curr])
-        : null}
-    </Typography.Span>
-  )
+  const {navigationStrategy, linkProps} = options || {}
+
+  const mapValueToDisplay = value =>
+    navigationStrategy?.DetailLink ? (
+      <span onClick={e => e.stopPropagation()}>
+        <navigationStrategy.DetailLink entityKey={value.key} {...linkProps}>
```

Comment:
(maybe place `entityKey` prop after `...linkProps` to prevent it from getting overridden accidentally)